src/Components/Dropdowns/Composed/TypeAheadDropDown.tsx

```diff
@@ -94,28 +66,42 @@ export const TypeAheadDropDown: FC<OwnProps> = ({
 
   const [selectIndex, setSelectIndex] = useState(-1)
   const [queryResults, setQueryResults] = useState(items)
-  const [menuOpen, setMenuOpen] = useState<MenuStatus>(MenuStatus.Closed)
-
+  const [menuStatus, setMenuStatus] = useState<MenuStatus>(MenuStatus.Closed)
+  const [userHasTyped, setUserHasTyped] = useState(false)
   const [selectedItem, setSelectedItem] = useState<SelectableItem | null>(
     selectedOption
   )
 
-  let initialTypedValue = ''
+  let initialInputValue = ''
 
   if (selectedOption) {
-    initialTypedValue = getValueWithBackup(selectedOption.name, defaultNameText)
+    initialInputValue = selectedOption.name || defaultNameText
   }
 
-  const [typedValue, setTypedValue] = useState<string>(initialTypedValue)
+  const [inputValue, setInputValue] = useState<string>(initialInputValue)
+  const [backupSelectionValue, setBackupSelectionValue] = useState<string>(
+    initialInputValue
+  )
 
-  const largeListValidation =
-    largeListSearch && queryResults.length > largeListCeiling
+  useEffect(() => {
+    // If selectedOption is invalid (not in the list of items), then we want to clear the input value
```

Comment:
Does this happen? How would the user select an option that is not in the current list?

Reply:
selectedOption is a prop so they could technically send in an invalid default selection. I noticed this happened in the Ui. So I thought I could handle it here.